src/qwircl/core.cljs

```diff
@@ -33,34 +51,108 @@
   ; nothing yet
   state)
 
-(defn to-position [event]
-  ; x is easy, but for y we need to take the header
-  ; into account
-  (let [x (int (/ (:x event) size))
-        r (:y event)]
-    (if (<= r header)
-      {:in :hand :coordinates [x]}
-      {:in :grid :coordinates [x (int (/ (- r header) size))]}))) 
+(defn inside-dimensions? [xp yp {:keys [x y w h]}]
+  (and 
+   (<= x xp (+ x w))
+   (<= y yp (+ y h))))
+
+(defn get-clicked [x y]
+  (first (filter #(inside-dimensions? x y (% dimensions)) [:hand :grid :submit :undo])))
+
+(defn translate-hand [xp hand]
+  (let [x (int (/ (- xp (first header-translation)) size))]
+    (when (and (<= 0 x) (< x (count hand)))
+      [x])))
+
+(defn translate-grid [x y]
+  [(int (/ x size)) (int (/ (- y header) size))])
+
+(defn translate-event [state {:keys [x y]}]
+  (condp = (get-clicked x y)
```

Comment:
`condp =` can be `case` here. This might make sense as a multimethod too, e.g. `(defmulti translate-event [state event] (fn [_ {:keys [x y]}] (get-clicked x y)))`. That way the `(defmethod translate-event [state {:keys [x y]}] :hand 
...)` can be placed next to the consumer. It might make sense to take the refactoring further and combine this function with the consumer.

Reply:
Opened issue: https://github.com/ivansalazar/qwircl/issues/8